pop/service/api.go

```diff
@@ -0,0 +1,215 @@
+package service
+
+import (
+	"bytes"
+
+	"github.com/BurntSushi/toml"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/base64"
+	"github.com/satori/go.uuid"
+	"gopkg.in/dedis/onet.v1"
+	"gopkg.in/dedis/onet.v1/log"
+	"gopkg.in/dedis/onet.v1/network"
+)
+
+const (
+	// ErrorWrongPIN indicates a wrong PIN
+	ErrorWrongPIN = iota + 4100
+	// ErrorInternal indicates something internally went wrong - see the
+	// error message
+	ErrorInternal
+	// ErrorOtherFinals indicates that one or more of the other conodes
+	// are still missing the finalization-step
+	ErrorOtherFinals
+)
+
+func init() {
+	network.RegisterMessage(&FinalStatement{})
+	network.RegisterMessage(&PopDesc{})
+}
+
+// FinalStatement is the final configuration holding all data necessary
+// for a verifier.
+type FinalStatement struct {
+	// Desc is the description of the pop-party.
+	Desc *PopDesc
+	// Attendees holds a slice of all public keys of the attendees.
+	Attendees []abstract.Point
+	// Signature is created by all conodes responsible for that pop-party
+	Signature []byte
+}
+
+// The toml-structure for (un)marshaling with toml
+type finalStatementToml struct {
+	Desc      *popDescToml
+	Attendees []string
+	Signature string
+}
+
+// NewFinalStatementFromString creates a final statement from a string.
+func NewFinalStatementFromString(s string) *FinalStatement {
+	fsToml := &finalStatementToml{}
+	_, err := toml.Decode(s, fsToml)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	sis := []*network.ServerIdentity{}
+	for _, s := range fsToml.Desc.Roster {
+		uid, err := uuid.FromString(s[2])
+		if err != nil {
+			log.Error(err)
+			return nil
+		}
+		sis = append(sis, &network.ServerIdentity{
+			Address:     network.Address(s[0]),
+			Description: s[1],
+			ID:          network.ServerIdentityID(uid),
+			Public:      B64ToPoint(s[3]),
+		})
+	}
+	rostr := onet.NewRoster(sis)
+	desc := &PopDesc{
+		Name:     fsToml.Desc.Name,
+		DateTime: fsToml.Desc.DateTime,
+		Location: fsToml.Desc.Location,
+		Roster:   rostr,
+	}
+	atts := []abstract.Point{}
+	for _, p := range fsToml.Attendees {
+		atts = append(atts, B64ToPoint(p))
+	}
+	sig := make([]byte, 64)
+	sig, err = base64.StdEncoding.DecodeString(fsToml.Signature)
+	if err != nil {
+		log.Error(err)
+		return nil
+	}
+	return &FinalStatement{
+		Desc:      desc,
+		Attendees: atts,
+		Signature: sig,
+	}
+}
+
+// PointToB64 converts an abstract.Point to a base64-point.
+func PointToB64(p abstract.Point) string {
```

Comment:
I find it clearly worth it and cleaner... It should be quite fast to do and merge since you already coded the most part. If you don't have time, I can try to make a quick one.